ci/scripts/check-links.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Checks links between files.
+
+set -e
+
+# Remove file containing invalid UTF-8
+rm -f sw/vendor/eembc_coremark/docs/html/index/General2.html
```

Comment:
Would there be an issue with `--exclude-path sw/vendor`? I don't think we include any of those docs in our own.

Reply:
`sw/vendor/eembc_coremark/docs/html/index/General2.html` isn't valid UTF-8. Even though we are using `--exclude-path sw/vendor` lychee still panics upon reading `General2.html`.

This file is still invalid UTF-8 [upstream](https://github.com/eembc/coremark), so we can't simply re-vendor it in. We could patch it or try to get it fixed upstream.

Reply:
Oh I see, that's a pain. I wonder why lychee is still reading the file if it's ignored, any ideas?

Anyway, I guess this is fine but it's a shame to have a script like this delete a file since you then can't run it outside the sandbox

Reply:
> We could patch it or try to get it fixed upstream.

This would be nice if you have time!

Reply:
> Thanks for getting started with this, automatic link checking will be very useful.
> 
> I agree with @rswarbrick making this easy to run manually would be good, maybe through Bazel's `quality` rules? I don't think we can run the script as is since it deletes a file.
> 
> Otherwise only nits, thanks!

@jwnrt 

It would be great to have this a part of Bazel's `quality` rules. If bazel is carrying out this check, it would be nice if Bazel manages the dependency to make it a little more hermetic. Although we could pull in a release archive with Bazel, in the future I'd like to add anchor checking. If I do, it may take a while to get this patch in upstream, so it would be good to be able to build our own version of lychee. So I think I will wait and see what people think the best way of including a patched rust binary is. We'll hopefully find this out in https://github.com/lowRISC/opentitan/pull/17996.

Until then do you mind if this check isn't part of Bazel's `quality` rules?